### PR TITLE
Enable Mamba as Summary Nets

### PR DESCRIPTION
**Pull request by stefanradev93**
Wraps Mamba State Space Model (SSM) as a powerful summary network for time series. Usage:

`
summary_net = MambaSSM()
`

Credit to: https://github.com/state-spaces/mamba/tree/main

Note: Only supports torch backend due to hardware-aware optimization.

### CODE REVIEW

File: bayesflow/wrappers/mamba.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        device : str, optional
+            The device to which the Mamba model is moved, typically "cuda" or "cpu" (default is "cuda").
+        **kwargs : dict
+            Additional keyword arguments passed to the `keras.layers.Layer` initializer.
+        """
+
+        super().__init__(**kwargs)
+
+        if keras.backend.backend() != "torch":
+            raise EnvironmentError("Mamba is only available using torch backend.")
```
Comment (LarsKue) [inline bayesflow/wrappers/mamba.py:67]: `EnvironmentError` is deprecated in favor of `OSError` which does not really fit here. Let's use a generic `RuntimeError` instead.

File: bayesflow/wrappers/mamba.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            Minimum delta time for Mamba (default is 0.001).
+        dt_max : float, optional
+            Maximum delta time for Mamba (default is 0.1).
+        device : str, optional
+            The device to which the Mamba model is moved, typically "cuda" or "cpu" (default is "cuda").
+        **kwargs : dict
+            Additional keyword arguments passed to the `keras.layers.Layer` initializer.
+        """
+
+        super().__init__(**kwargs)
```
Comment (LarsKue) [inline bayesflow/wrappers/mamba.py:64]: Should filter with `keras_kwargs` (or `layer_kwargs` in my upcoming PR)

File: bayesflow/wrappers/mamba.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            feature_dim,
+            kernel_size=1,
+            strides=1,
+        )
+        self.layer_norm = keras.layers.LayerNormalization()
+
+    def call(self, x: Tensor, training: bool = False, **kwargs) -> Tensor:
+        out_forward = self._call(x, training=training, **kwargs)
+        if self.bidirectional:
+            out_backward = self._call(keras.ops.flip(x, axis=1), training=training, **kwargs)
```
Comment (LarsKue) [inline bayesflow/wrappers/mamba.py:85]: Should this flip on axis 1 or -1?

File: bayesflow/wrappers/mamba.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        return out
+
+    @sanitize_input_shape
+    def build(self, input_shape):
+        super().build(input_shape)
+        self.call(keras.ops.zeros(input_shape))
+
+
+@serializable("bayesflow.wrappers")
+class MambaSSM(SummaryNetwork):
```
Comment (LarsKue) [inline bayesflow/wrappers/mamba.py:102]: I would be in favor of splitting the `mamba.py` file into one for each class.

File: bayesflow/wrappers/mamba.py [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+        device : str, optional
+            The computing device. Currently, only "cuda" is supported (default is "cuda").
+        **kwargs : dict
+            Additional keyword arguments passed to the `SummaryNetwork` parent class.
+        """
+
+        super().__init__(**kwargs)
+
+        if device != "cuda":
+            raise NotImplementedError("MambaSSM only supports cuda as `device`.")
```
Comment (LarsKue) [inline bayesflow/wrappers/mamba.py:160]: Why make this an argument if there is no other choice?

**Review by LarsKue — COMMENTED** [review]
Looks good overall. I would change some minor things, see individual comments.

File: bayesflow/wrappers/mamba.py (hunk 3 above)
Comment (stefanradev93) [reply bayesflow/wrappers/mamba.py:85]: Axis 1 -> time axis.

File: bayesflow/wrappers/mamba.py (hunk 5 above)
Comment (stefanradev93) [reply bayesflow/wrappers/mamba.py:160]: May change in future.

File: bayesflow/wrappers/mamba.py (hunk 3 above)
Comment (stefanradev93) [reply bayesflow/wrappers/mamba.py]: Robustified via axis=-2

File: bayesflow/wrappers/mamba.py (hunk 2 above)
Comment (stefanradev93) [reply bayesflow/wrappers/mamba.py]: done

File: bayesflow/wrappers/mamba.py (hunk 1 above)
Comment (stefanradev93) [reply bayesflow/wrappers/mamba.py]: done

**Review by LarsKue — APPROVED** [review]
